src/vision/frc_vision_final.py

```diff
@@ -78,106 +111,119 @@ def newLine(filtered_lines, new_line, filtered_line_img, x1, y1, x2, y2, X_TOTAL
+                            FILTERED_LINES, NEW_LINE, FILTERED_LINE_IMG, x1, y1, x2, y2, X_TOTAL, Y_TOTAL)
+                elif new_slope < -ANGLE_THRESHOLD or new_slope > ANGLE_THRESHOLD:
+                    X_TOTAL, Y_TOTAL = newLine(FILTERED_LINES, NEW_LINE,
+                        FILTERED_LINE_IMG, x1, y1, x2, y2, X_TOTAL, Y_TOTAL)
+    
+            NUM_LINES=len(FILTERED_LINES)
             if FILTERED_LINES:
-                if (new_slope < -ANGLE_THRESHOLD or new_slope > ANGLE_THRESHOLD) and unequal(new_slope, FILTERED_LINES):
-                    X_TOTAL, Y_TOTAL = newLine(
-                        FILTERED_LINES, NEW_LINE, FILTERED_LINE_IMG, x1, y1, x2, y2, X_TOTAL, Y_TOTAL)
-            elif new_slope < -ANGLE_THRESHOLD or new_slope > ANGLE_THRESHOLD:
-                X_TOTAL, Y_TOTAL = newLine(FILTERED_LINES, NEW_LINE,
-                    FILTERED_LINE_IMG, x1, y1, x2, y2, X_TOTAL, Y_TOTAL)
-
-        NUM_LINES=len(FILTERED_LINES)
-        if FILTERED_LINES:
-            X_AVG=0
-            Y_AVG=0
-
-            if len(VALS) == POINT_SAMPLES:
-                VALS.pop(0)
-                VALS.append([X_TOTAL/(2*NUM_LINES), Y_TOTAL/(2*NUM_LINES)])
-
-                # TODO: Test if you could keep this continuous, not re-add everything every loop
-                # Subtract Popped val, add new val?
-                for VAL in VALS:
-                    X_VAL, Y_VAL = VAL
-                    X_AVG += X_VAL
-                    Y_AVG += Y_VAL
-
-                X_AVG=int(X_AVG / POINT_SAMPLES)
-                Y_AVG=int(Y_AVG / POINT_SAMPLES)
-
-                offset = 2 * (X_AVG - (WIDTH/2)) / WIDTH
-                cv2.circle(FILTERED_LINE_IMG, (X_AVG, Y_AVG),
-                           5, [255, 255, 255], -1)
-                try:
-                    dist_to_target = depth.get_distance(X_AVG, Y_AVG)
-                except:
-                    None
-
-                # Smart Dashboard variables
-                sd.putBoolean("Sees Target", True)
-                sd.putNumber("Ratio Offset", offset)
-                sd.putNumber("Angle Offset", PIXEL_ANGLE * X_AVG)
-                sd.putNumber("Distance To Target", dist_to_target)
-
-            else:
-                VALS.append([X_TOTAL/(2*NUM_LINES), Y_TOTAL/(2*NUM_LINES)])
-
-        for LINE in LINES:
-            x1, y1, x2, y2=LINE[0]
-            cv2.line(LINE_IMG, (x1, y1), (x2, y2), (0, 255, 0), 1)
+                X_AVG=0
+                Y_AVG=0
+    
+                if len(VALS) == POINT_SAMPLES:
+                    VALS.pop(0)
+                    VALS.append([X_TOTAL/(2*NUM_LINES), Y_TOTAL/(2*NUM_LINES)])
+    
+                    # TODO: Test if you could keep this continuous, not re-add everything every loop
+                    # Subtract Popped val, add new val?
+                    for VAL in VALS:
+                        X_VAL, Y_VAL = VAL
+                        X_AVG += X_VAL
+                        Y_AVG += Y_VAL
```

Comment:
You should be able to split the X and Y values with something from: https://stackoverflow.com/questions/7558908/unpacking-a-list-tuple-of-pairs-into-two-lists-tuples

Then you can use `sum(x_list)`. Or you could get even fancier and use the built in `mean` function. https://appdividend.com/2019/01/28/python-statistics-tutorial-mean-function-example/